pkg/apis/k0s/v1beta1/clusterconfig_types.go

```diff
@@ -332,6 +336,64 @@ func (s *ClusterSpec) Validate() (errs []error) {
 		errs = append(errs, err)
 	}
 
+	for _, err := range s.validateKine(online) {
+		errs = append(errs, err)
+	}
+
+	return
+}
+
+func (s *ClusterSpec) validateKine(online bool) (errs field.ErrorList) {
+	if s.Storage.Type != KineStorageType {
+		return
+	}
+
+	if s.Storage.Kine == nil {
+		errs = append(errs, field.NotFound(field.NewPath("storage", "kine"), "missing"))
+		return
+	}
+
+	dataSource := s.Storage.Kine.DataSource
+
+	if dataSource == "" {
```

Comment:
I wasn't sure if this can happen. But if anyone can give me a hint.

I am basing my assumption on the following:

```yaml
       storage:
          type: kine
```

The above currently enables sqlite with defaults. But are the defaults available here?